utilities/transfer-cards/main.go

```diff
@@ -43,7 +48,87 @@ func getProject(client *github.Client, ctx context.Context, projectName string)
 		}
 		opt.Page = resp.NextPage
 	}
-	return nil, fmt.Errorf("project '%s' not found in repo %s/%s", projectName, *repoOwner, *repoName)
+
+	// don't want to create a project if the project is a source project
+	if source {
+		return nil, fmt.Errorf("project '%s' not found in repo %s/%s", projectName, *repoOwner, *repoName)
+	}
+	// if this is a dry run do not create the project
+	if *dryrun {
+		return nil, fmt.Errorf("project '%s' not found in repo %s/%s if you would like the utility to create the project rerun without the dryrun option\n", projectName, *repoOwner, *repoName)
+	}
+
+	// if project was not found try creating the project
+	if getProjectErr != nil || project == nil {
+		log.Print("Project not found creating project\n")
+		project, createProjectErr := createProject(client, ctx, projectName)
+		if createProjectErr != nil {
+			return nil, createProjectErr
+		} else {
+			log.Print("Project creation successful\n")
+			return project, nil
+		}
+	} else {
+
+		return project, nil
+	}
+}
+
+// When a project is created the release bot needs to add project columns (triage, cherry-pick, and
+// cherry-picked) to the project. The transfer card utility should not progress until the release-bot is done
+func releaseBotDone(client *github.Client, ctx context.Context, projectID int) (bool, error) {
+	columnsLength := 0
+	retries := 1
+
+	for {
+		log.Printf("Release bot progress: retries %d project columns:  %d\n", retries, columnsLength)
+		if retries == 3 || columnsLength == 3 {
+			break
+		}
+
+		columns, _, err := client.Projects.ListProjectColumns(context.Background(), projectID, &github.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+		columnsLength = len(columns)
+		retries += 1
+
+		time.Sleep(5 * time.Second)
+	}
+
+	if columnsLength == 3 {
+		return true, nil
+	} else {
+		return false, nil
+	}
+}
+
+func createProject(client *github.Client, ctx context.Context, projectName string) (*github.Project, error) {
+	opt := &github.ProjectOptions{State: "all"}
+	opt.Name = projectName
+	opt.State = "open"
+	info := strings.Split(projectName, "-") //ex. 18.02.0-ce-rc2 -> [18.02.0, ce, rc2]
+	var body string
+	body = ""
+	if len(info) == 3 {
+		body = fmt.Sprintf(`Docker %s %s %s release`, info[0], strings.ToUpper(info[1]), strings.ToUpper(info[2]))
+	}
+	opt.Body = body
```

Comment:
Why not just do this when you instantiate `opt`?

Reply:
This part expects a name in a certain format, so I'm only filling out the body of the project description if the project name matches that format.